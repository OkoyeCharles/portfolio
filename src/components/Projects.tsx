import React, { useState } from "react";
import ProjectGrid from "./ProjectGrid";
import projects from "../data/projects";
import ProjectPopup from "./ProjectPopup";
import { Reveal } from "react-awesome-reveal";
import { keyframes } from "@emotion/react";

const slideUp = keyframes`
  from {
    transform: translateY(2em);
    opacity: 0;
  }
  to {
    transform: translateY(0);
    opacity: 1;
  }
`;

interface ProjectsProps {
  mediaWidth: number;
}

const Projects: React.FC<ProjectsProps> = ({ mediaWidth }) => {
  const [popupIndex, setPopupIndex] = useState<number>(0);
  const [popupDisplayed, setPopupDisplayed] = useState<boolean>(false);

  const togglePopup = (index: number, active: boolean) => {
    setPopupIndex(index);
    setPopupDisplayed(active);
  };

  return (
    <section className="projects">
      <div className="projects__container">
        <h2 className="projects__header">Projects</h2>
        <Reveal triggerOnce fraction={1} delay={100} keyframes={slideUp}>
          <div className="projects__coa">
            I'm a <strong>creative</strong> developer who loves to work on{" "}
            <strong>excellent and appealing websites</strong> with a passion for
            UI/UX... always yielding high returns on what is invested. Below
            showcases some of my latest projects.
          </div>
        </Reveal>
        <ProjectGrid
          projects={projects}
          mediaWidth={mediaWidth}
          togglePopup={togglePopup}
        />
        <ProjectPopup
          popupDisplayed={popupDisplayed}
          mediaWidth={mediaWidth}
          index={popupIndex}
          togglePopup={togglePopup}
        />
      </div>
    </section>
  );
};

export default Projects;
