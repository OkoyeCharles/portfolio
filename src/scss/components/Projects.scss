@use "../abstracts" as *;

.projects {
  background: $dark;
}

.projects * {
  color: $light;
}

.projects__container {
  max-width: 1600px;
  margin-inline: auto;
  padding: 1em 1.8em 5em 4.5em;
}

.projects__header {
  width: fit-content;
  font-size: 1.5em;
  position: relative;
  text-transform: uppercase;
  margin-bottom: 2em;
}

.projects__header::after {
  position: absolute;
  content: "";
  bottom: -0.25em;
  left: 0.5rem;
  width: 100%;
  height: 0.75em;
  background: transparentize($color: $light, $amount: 1 - 0.125);
}

.projects__coa {
  margin-bottom: 2em;
  font-size: 1.375em;
  font-weight: 300;
  color: transparentize($color: $light, $amount: 1 - 0.75);
}

.projects__coa strong {
  color: transparentize($color: $light, $amount: 1 - 0.9);
}

@media screen and (max-width: 700px) {
  .projects__container {
    padding: 1.8em;
  }

  .projects__header {
    font-size: clamp(1.25em, 4vw, 1.5em);
  }

  .projects__coa {
    font-size: clamp(1em, 3.5vw, 1.25em);
    text-align: center;
  }
}
