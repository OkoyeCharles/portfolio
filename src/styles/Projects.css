.projects {
  background: #111;
  min-height: 100vh;
}

.projects * {
  color: #fff;
}

.projects__container {
  max-width: 1600px;
  margin-inline: auto;
  padding: 1em 1.8em 0 4.5em;
}

.projects__header {
  width: fit-content;
  font-size: 1.5em;
  position: relative;
  text-transform: uppercase;
  margin-bottom: 2em;
}

.projects__header::after {
  position: absolute;
  content: '';
  bottom: -0.25em;
  left: 0.5rem;
  width: 100%;
  height: 0.75em;
  background: rgba(255, 255, 255, 0.125);
}

.projects__coa {
  margin-bottom: 2em;
  font-size: 1.375em;
  font-weight: 300;
  color: rgba(255, 255, 255, 0.75);
}

.projects__coa strong {
  color: rgba(255, 255, 255, 0.9);
}