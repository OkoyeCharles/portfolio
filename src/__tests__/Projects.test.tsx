import renderer from 'react-test-renderer';
import Projects from '../components/Projects';

it('<Projects /> component renders properly', () => {
  const component = renderer.create(
    <Projects mediaWidth={1000}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});