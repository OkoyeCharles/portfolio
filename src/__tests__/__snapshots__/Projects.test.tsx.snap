// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Projects /> component renders properly 1`] = `
<section
  className="projects"
>
  <a
    id="projects"
  />
  <div
    className="projects__container"
  >
    <h2
      className="projects__header"
    >
      Projects
    </h2>
    <div
      className="css-sanqhm"
      style={
        Object {
          "animationDelay": "100ms",
        }
      }
    >
      <div
        className="projects__coa"
        style={Object {}}
      >
        I'm a 
        <strong>
          creative
        </strong>
         developer who loves to work on
         
        <strong>
          excellent and appealing websites
        </strong>
         with a passion for UI/UX... always yielding high returns on what is invested. Below showcases some of my latest projects.
      </div>
    </div>
    <ul
      className="projectGrid"
    >
      <li
        className="projectCard css-sanqhm"
        style={
          Object {
            "animationDelay": "0ms",
            "background": "url(amazon-clone.webp)",
          }
        }
      >
        <div
          className="projectCard__title"
        >
          <h3>
            Amazon Clone
          </h3>
          <button
            aria-label="Open Project"
            className="projectCard__arrow"
            onClick={[Function]}
            role="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
              />
            </svg>
          </button>
        </div>
        <div
          className="projectCard__note"
        >
          A full-stack e-commerce website inspired by amazon.com
        </div>
      </li>
      <li
        className="projectCard css-sanqhm"
        style={
          Object {
            "animationDelay": "0ms",
            "background": "url(lavish-cuisine.webp)",
          }
        }
      >
        <div
          className="projectCard__title"
        >
          <h3>
            Lavish Cuisine
          </h3>
          <button
            aria-label="Open Project"
            className="projectCard__arrow"
            onClick={[Function]}
            role="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
              />
            </svg>
          </button>
        </div>
        <div
          className="projectCard__note"
        >
          A meal-based website that gives you all the information you need about your favourite meals.
        </div>
      </li>
      <li
        className="projectCard css-sanqhm"
        style={
          Object {
            "animationDelay": "0ms",
            "background": "url(nethub.webp)",
          }
        }
      >
        <div
          className="projectCard__title"
        >
          <h3>
            Nethub
          </h3>
          <button
            aria-label="Open Project"
            className="projectCard__arrow"
            onClick={[Function]}
            role="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
              />
            </svg>
          </button>
        </div>
        <div
          className="projectCard__note"
        >
          A Movie App that extracts information from an external API.
        </div>
      </li>
      <li
        className="projectCard css-sanqhm"
        style={
          Object {
            "animationDelay": "0ms",
            "background": "url(save-a-child.webp)",
          }
        }
      >
        <div
          className="projectCard__title"
        >
          <h3>
            Save A Child
          </h3>
          <button
            aria-label="Open Project"
            className="projectCard__arrow"
            onClick={[Function]}
            role="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
              />
            </svg>
          </button>
        </div>
        <div
          className="projectCard__note"
        >
          A website built with JavaScript... focused on preserving the life of children who suffered during the COVID-19 outbreak.
        </div>
      </li>
    </ul>
    <div
      className="projectPopup"
    >
      <div
        className="projectPopup__container"
      >
        <div
          className="projectPopup__close"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
            />
          </svg>
        </div>
        <div
          className="projectPopup__content"
        >
          <h2
            className="projectPopup__title"
          >
            Amazon Clone
            <a
              aria-label="Move to live website"
              className="projectPopup__titleLink"
              href="https://clone-f50ae.web.app/"
              rel="noreferrer-noopener"
              target="_blank"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                version="1.1"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6.879 9.934c-0.208 0-0.416-0.079-0.575-0.238-1.486-1.486-1.486-3.905 0-5.392l3-3c0.72-0.72 1.678-1.117 2.696-1.117s1.976 0.397 2.696 1.117c1.486 1.487 1.486 3.905 0 5.392l-1.371 1.371c-0.317 0.317-0.832 0.317-1.149 0s-0.317-0.832 0-1.149l1.371-1.371c0.853-0.853 0.853-2.241 0-3.094-0.413-0.413-0.963-0.641-1.547-0.641s-1.134 0.228-1.547 0.641l-3 3c-0.853 0.853-0.853 2.241 0 3.094 0.317 0.317 0.317 0.832 0 1.149-0.159 0.159-0.367 0.238-0.575 0.238z"
                />
                <path
                  d="M4 15.813c-1.018 0-1.976-0.397-2.696-1.117-1.486-1.486-1.486-3.905 0-5.392l1.371-1.371c0.317-0.317 0.832-0.317 1.149 0s0.317 0.832 0 1.149l-1.371 1.371c-0.853 0.853-0.853 2.241 0 3.094 0.413 0.413 0.962 0.641 1.547 0.641s1.134-0.228 1.547-0.641l3-3c0.853-0.853 0.853-2.241 0-3.094-0.317-0.317-0.317-0.832 0-1.149s0.832-0.317 1.149 0c1.486 1.486 1.486 3.905 0 5.392l-3 3c-0.72 0.72-1.678 1.117-2.696 1.117z"
                />
              </svg>
            </a>
          </h2>
          <div
            className="projectPopup__stack"
          >
            <span>
              React
            </span>
            <span>
              Redux
            </span>
            <span>
              Firebase
            </span>
            <span>
              Sass
            </span>
          </div>
          <p
            className="projectPopup__description"
          >
            A Responsive Full-stack e-commerce website inspired by the Amazon website. Built with React, Redux, and Firebase. Including functionalities such as Creating/Signing in to an account, Adding items to the cart, Making payments and so on
          </p>
          <div
            className="projectPopup__image"
          >
            <img
              alt="Amazon Clone"
              src="amazon-clone.webp"
            />
          </div>
        </div>
        <div
          className="projectPopup__links"
        >
          <a
            href="https://clone-f50ae.web.app/"
            rel="noreferrer-noopener"
            target="_blank"
          >
            <button>
              View Live 
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M14 13.9633H16V7.96331H10V9.96331H12.5858L7.25623 15.2929L8.67044 16.7071L14 11.3775V13.9633Z"
                  fill="currentColor"
                />
                <path
                  clipRule="evenodd"
                  d="M23 19C23 21.2091 21.2091 23 19 23H5C2.79086 23 1 21.2091 1 19V5C1 2.79086 2.79086 1 5 1H19C21.2091 1 23 2.79086 23 5V19ZM19 21H5C3.89543 21 3 20.1046 3 19V5C3 3.89543 3.89543 3 5 3H19C20.1046 3 21 3.89543 21 5V19C21 20.1046 20.1046 21 19 21Z"
                  fill="currentColor"
                  fillRule="evenodd"
                />
              </svg>
            </button>
          </a>
          <a
            href="https://github.com/OkoyeCharles/amazon-clone"
            rel="noreferrer-noopener"
            target="_blank"
          >
            <button>
              View Source 
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
                />
              </svg>
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
